src/store/reducers/messagesReducer.js

```diff
@@ -19,6 +20,7 @@ export default function (storage) {
 
   return function reducer(state = initialState, action) {
     const storeMessage = storeMessageTo(storage);
+    const clear = clearMessages(storage);
```

Comment:
is it supposed to be inside the case condition for `actionTypes.CLEAR`? if it's outside it will clear messages in other cases as well.

Reply:
this line just saves the function `clearMessages` in this `clear` variable, 
the `clearMessages` definition is: `export const clearMessages = storage => () => { ... }`
so, when `clearMessages(storage)` is called it returns the function that erase the messages. and this function is called with `clear()`

source: https://stackoverflow.com/questions/32782922/what-do-multiple-arrow-functions-mean-in-javascript

Reply:
Oh, I didn't notice.